src/ui/SmartSnippet/SmartSnippet.ts

```diff
@@ -60,19 +68,45 @@ export class SmartSnippet extends Component {
   }
 
   public createDom() {
-    this.buildShadow();
-    this.buildSourceContainer();
+    this.element.appendChild(this.buildAnswerContainer());
+    this.element.appendChild(
+      new UserFeedbackBanner(
+        isUseful =>
+          this.usageAnalytics.logCustomEvent<IAnalyticsNoMeta>(
+            isUseful ? analyticsActionCauseList.likeSmartSnippet : analyticsActionCauseList.dislikeSmartSnippet,
+            {},
+            this.element,
+            this.lastRenderedResult
+          ),
+        explaination => console.info(`Explaination: ${explaination}`)
+      ).build()
+    );
+  }
+
+  private buildAnswerContainer() {
+    return (this.answerContainer = $$(
+      'div',
+      {
+        className: ANSWER_CONTAINER_CLASSNAME
+      },
+      this.buildShadow(),
+      this.buildSourceContainer()
+    ).el);
   }
 
   private buildShadow() {
-    this.element.appendChild((this.shadowContainer = $$('div', { className: SHADOW_CLASSNAME }).el));
+    this.shadowContainer = $$('div', { className: SHADOW_CLASSNAME }).el;
     const shadow = this.shadowContainer.attachShadow({ mode: 'open' });
     shadow.appendChild((this.snippetContainer = $$('section', { className: CONTENT_CLASSNAME }).el));
-    shadow.appendChild(this.buildStyle());
+    const style = this.buildStyle();
+    if (style) {
+      shadow.appendChild(style);
+    }
+    return this.shadowContainer;
   }
 
   private buildSourceContainer() {
-    this.element.appendChild((this.sourceContainer = $$('div', { className: SOURCE_CLASSNAME }).el));
+    return (this.sourceContainer = $$('div', { className: SOURCE_CLASSNAME }).el);
```

Comment:
The `this. sourceContainer ` seems to be only assigned to, but never read from. Could we remove it ?

Reply:
`this.sourceContainer` is read in `renderSource`, I can't remove it.